File: Kattis/wheresmyinternet.cpp
#include <bits/stdc++.h>
using namespace std;

const int N = 2e5 + 5;

int n, m;
bool vis[N];
vector<int> adj[N];

void dfs(int v) {
  vis[v] = true;
  for (int u : adj[v]) {
    if (!vis[u]) {
      dfs(u);
    }
  }
}

int main() {
  cin >> n >> m;
  for (int i = 0 ; i < m; ++i) {
    int a, b;
    cin >> a >> b;
    adj[a].push_back(b);
    adj[b].push_back(a);
  }
  dfs(1);
  vector<int> ans;
  for (int i = 1; i <= n; ++ i) {
    if (!vis[i]) {
      ans.push_back(i);
    }
  }
  if (ans.empty()) {
    cout << "Connected";
  } else {
    for (int v : ans) {
      cout << v << '\n';
    }
  }
}

File: Kattis/wheresmyinternet.py
n, m = map(int, input().split())

adj = []
for i in range(n):
  adj.append([])

vis = [False for _ in range(n)]

for _ in range(m):
  a, b = map(int, input().split())
  a -= 1
  b -= 1
  adj[a].append(b)
  adj[b].append(a)

# print(adj)

def dfs(v: int) -> None:
  vis[v] = True
  for x in adj[v]:
    if not vis[x]:
      dfs(x)

dfs(0)

answer = []
for i in range(n):
  if not vis[i]:
    answer.append(str(i + 1))

if len(answer) == 0:
  print('Connected')
else:
  print('\n'.join(answer))
